src/appleseed/renderer/kernel/rendering/masterrenderer.cpp

```diff
@@ -461,13 +462,24 @@ struct MasterRenderer::Impl
         // Updating the trace context causes ray tracing acceleration structures to be updated or rebuilt.
         m_project.update_trace_context();
 
+        // Load the checkpoint if any.
+        Frame& frame = *m_project.get_frame();
+
+        PermanentShadingResultFrameBufferFactory* buffer_factory =
```

Comment:
What guarantees that the shading result framebuffer is a permanent one here?

Reply:
The code that generate the renderer and the buffer :D You think I should dynamic cast and test the result ?